crates/chia-wallet/src/puzzles/singleton.rs

```diff
@@ -21,8 +21,8 @@ pub struct SingletonStruct {
 
 #[derive(Debug, Clone, PartialEq, Eq, ToClvm, FromClvm)]
 #[clvm(list)]
-pub struct SingletonSolution<I> {
-    pub proof: Proof,
+pub struct SingletonSolution<I, P = LineageProof> {
```

Comment:
doesn't the proof for a singleton always have to be a lineage proof?

Reply:
When you mint an NFT, create a DID, or launch any other singleton, there's one coin where it's an EveProof